core/src/main/java/org/apache/iceberg/view/ViewVersionReplace.java

```diff
@@ -0,0 +1,138 @@
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT;
+
+import java.util.List;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.Tasks;
+
+public class ViewVersionReplace implements ReplaceViewVersion {
+  private final ViewOperations ops;
+  private final List<ViewRepresentation> viewRepresentationsToAdd = Lists.newArrayList();
+  private ViewMetadata base;
+  private Namespace defaultNamespace;
+  private String defaultCatalog;
+  private Schema schema;
+
+  ViewVersionReplace(ViewOperations ops) {
+    this.ops = ops;
+    this.base = ops.current();
+  }
+
+  @Override
+  public ViewVersion apply() {
+    this.base = ops.refresh();
+
+    ViewVersion viewVersion = base.currentVersion();
+
+    if (!viewRepresentationsToAdd.isEmpty()) {
+      long timestampMillis = System.currentTimeMillis();
+      int newVersion = viewVersion.versionId() + 1;
+
+      return ImmutableViewVersion.builder()
+          .versionId(newVersion)
+          .timestampMillis(timestampMillis)
+          .schemaId(null != schema ? schema.schemaId() : viewVersion.schemaId())
+          .defaultNamespace(defaultNamespace)
+          .defaultCatalog(defaultCatalog)
+          .putSummary("operation", "replace")
+          .addAllRepresentations(viewRepresentationsToAdd)
+          .build();
+    }
+
+    return viewVersion;
+  }
+
+  @Override
+  public void commit() {
+    Tasks.foreach(ops)
+        .retry(
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_NUM_RETRIES, COMMIT_NUM_RETRIES_DEFAULT))
+        .exponentialBackoff(
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_MIN_RETRY_WAIT_MS, COMMIT_MIN_RETRY_WAIT_MS_DEFAULT),
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_MAX_RETRY_WAIT_MS, COMMIT_MAX_RETRY_WAIT_MS_DEFAULT),
+            PropertyUtil.propertyAsInt(
+                base.properties(), COMMIT_TOTAL_RETRY_TIME_MS, COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT),
+            2.0 /* exponential */)
+        .onlyRetryOn(CommitFailedException.class)
+        .run(
+            taskOps -> {
+              ViewVersion newVersion = apply();
+              // nothing to do if the version didn't change
+              if (this.base.currentVersion().equals(newVersion)) {
+                return;
+              }
+
+              ViewMetadata.Builder builder = ViewMetadata.buildFrom(this.base);
+
+              if (null != schema) {
+                builder.addSchema(schema);
```

Comment:
Need to make sure schema is handled correctly here as well.

Reply:
fixed